core/storage/topic/gae_models.py

```diff
@@ -21,6 +21,104 @@
+    name = ndb.StringProperty(required=True, indexed=True)
+    # This includes additional metadata like the description of the topic and
+    # some content paragraphs.
+    description = ndb.StringProperty(indexed=False)
+    # This consists of the list of canonical story ids that are part of
+    # this topic.
+    canonical_story_ids = ndb.StringProperty(repeated=True, indexed=False)
+    # This consists of the list of additional (non-canonical) story ids that
+    # are part of this topic.
+    additional_story_ids = ndb.StringProperty(repeated=True, indexed=False)
+    # This consists of the full list of skill ids that are part of this topic.
+    skill_ids = ndb.StringProperty(repeated=True, indexed=False)
+    # The ISO 639-1 code for the language this question is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+
+
+class TopicCommitLogEntryModel(base_models.BaseCommitLogEntryModel):
+    """Log of commits to topics.
+
+    A new instance of this model is created and saved every time a commit to
+    TopicModel occurs.
+
+    The id for this model is of the form
+    'topic-{{TOPIC_ID}}-{{TOPIC_VERSION}}'.
+    """
+    # The id of the topic being edited.
+    topic_id = ndb.StringProperty(indexed=True, required=True)
+
+    @classmethod
+    def get_commit(cls, topic_id, version):
+        """Returns the commit corresponding to the given topic id and
+        version number.
+
+        Args:
+            topic_id: str. The id of the topic being edited.
+            version: int. The version number of the topic after the commit.
+
+        Returns:
+            The commit with the given topic id and version number.
+        """
+        return cls.get_by_id('topic-%s-%s' % (topic_id, version))
+
+
+class TopicSummaryModel(base_models.BaseModel):
+    """Summary model for an Oppia Topic.
+
+    This should be used whenever the content blob of the topic is not
+    needed (e.g. search results, etc).
+
+    A TopicSummaryModel instance stores the following information:
+
+        id, description, language_code, last_updated, created_on, version.
+
+    The key of each instance is the topic id.
+    """
+
+    # The name of the topic.
+    name = ndb.StringProperty(required=True, indexed=True)
+    # The ISO 639-1 code for the language this collection is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+
+    # Time when the topic model was last updated (not to be
+    # confused with last_updated, which is the time when the
+    # topic *summary* model was last updated).
+    topic_model_last_updated = ndb.DateTimeProperty(required=True, indexed=True)
+    # Time when the topic model was created (not to be confused
+    # with created_on, which is the time when the topic *summary*
+    # model was created).
+    topic_model_created_on = ndb.DateTimeProperty(required=True, indexed=True)
+    # The number of stories (both canonical and additional) that are part
```

Comment:
Can we have: canonical_story_count, additional_story_count? Seems reasonable to differentiate the two since the former is a sort of requirement and the latter is optional.

Reply:
Done.